### ISSUE TITLE
로밍 엣지 추론 스케줄링 방안 탐구

### ISSUE THREAD

**Issue by jungae-park** [ddps-lab/edge-management-module#6]
* 추론 가능한 장비의 여러 컨테이너에 DNN 모델 전달 및 추론을 위한 스케줄링
https://encouraging-lantern-5d1.notion.site/0f114418fab74b43b334cbb9df8ab109

**Comment by hyu-leeky** [comment]
그래 정애야 좋다. 가능한만큼 자세히 내용을 구상해보고 다음주에 같이 만나서 이야기를 나누어보자.

**Comment by jungae-park** [comment]
아래와 같은 단계로 스케줄링 구현을 위한 작업을 진행해보려고 합니다.

1. 크고 작은 여러 워크로드 시나리오 생성
* 논문에서 사용된 다양한 워크로드 확인 및 재현
2. 다양한 워크로드를 서빙 할 수 있는 스케줄링 구현
* 컨테이너 스케줄링 관련 기존 기술 확인
* 스케줄링에 사용 할 feature 선택
(cpu 사용량, memory 사용량, gpu 사용량, IPS, inference time, Flops/parameter, container counts ... etc)

**Comment by jungae-park** [comment]
1. 실제 환경에서 CNN 모델 추론을 위한 구성하여 모델 서빙 및 추론
    - k8s cluster로 구성되어 있는 엣지 노드의 hw spec 파악
        - cpu, memory, gpu, power
    - 여러 크고 작은 모델을 무작위로 노드에 전달
        - 아무 옵션 없이 여러개 크기의 pod-container 생성
        - container에서 추론 진행 후 metric 성능 확인
        - 성능 패턴 파악
    - 여러 크고 작은 모델을 파악한 패턴에 따라 기준을 세우고 적절한 노드에 전달
        - 여러 알고리즘에 값을 대입하여 계산하여 기준 생성
        - cpu, memory request limit을 두고 여러개의 pod-container 생성
        - 최대한 많은 양의 모델 서빙 및 추론

위와 같은 시나리오로 먼저 실험을 통해 워크로드를 일반화하여, 추후 트위터 데이터셋, 블록체인 데이터셋 같은 dynamic 워크로드 처리하는 시나리오도 실험을 진행하고자 합니다.
@kmu-leeky

**Comment by jungae-park** [comment]
@kmu-leeky 
네 교수님 실험은 저번에 미팅에서 말씀드린 것처럼, 위의 댓글처럼 우선 컨테이너의 크기를 달리해서 크고 작은 여러 모델을 무작위로 서빙하고 추론 가능한 장비와 모델의 패턴을 파악해보려고 합니다.

**Comment by hyu-leeky** [comment]
오케이. 그렇다면 발생하는 워크로드를 fix 해서 workload generator 를 먼저 구현하는 작업도 필요하겠다. 해당 이슈도 만들어서 함께 진행해보자. 어떤식의 워크로드를 만들건지에 대해서도 구체적으로 정리해보자.

**Comment by jungae-park** [comment]
넵 알겠습니다.
